assembly/src/assembler/instruction/crypto_ops.rs

```diff
@@ -158,8 +158,7 @@ pub(super) fn mtree_set(span: &mut SpanBuilder) -> Result<Option<CodeBlock>, Ass
 /// follows:
 /// - merged root, 4 elements
 ///
-/// This operation will fail if either of the input roots doesn't exist as Merkle tree in the
-/// advice provider.
+/// It is not expected for provided roots to be in the advice provider.
```

Comment:
I'm not sure that a brief comment is sufficient here, although it is the way I can make it _correct_ without additional explanations. 

I also considering this comment, but IMO it is a bit bulky:
```text
/// In case of working with Merkle trees, it is expected that the provided roots exist as Merkle 
/// trees in the advice provider, but this is not checked. In case of working with Merkle Mountain
/// range it is not expected for provided roots to be in the advice provider.
```

Reply:
IMO it is better to have the bulky comment. Otherwise it takes more time to onboard people.

Question: Why is there a difference in behavior? Can't we align it? It would be best if both structures behaved the same.

Reply:
I agree, that was an initial idea of the issue, although it turns out to be quite complicated to make this function work the same for the both structures: https://github.com/0xPolygonMiden/miden-vm/issues/943#issuecomment-1920178189